### DISCUSSION TITLE
Tutorial of AI Kit with Raspberry Pi 5 about YOLOv8n object detection | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1558] [Q&A]
# Tutorial of AI Kit with Raspberry Pi 5 about YOLOv8n object detection | Seeed Studio Wiki

This wiki demonstrates how to use yolov8n to do object detection with ai kit on raspberry pi5 from traing to deploying

https://wiki.seeedstudio.com/tutorial_of_ai_kit_with_raspberrypi5_about_yolov8n_object_detection/

**Comment by Atzingen** [comment]
Not clear on how to do the "hailomz parse ..." part with a custon yolo8 dataset. Does the dataset realy have to be converted to TFRecords and if so, how to use create_coco_tfrecord.py for this ?

**Reply by LJ-Hao** [reply]
You dont have to conver dataset to TFRecords, but TFRecord will accelate training. You can check this link:[Car Detection and Tracking System with RPi Hailo Al Kit](https://docs.edgeimpulse.com/experts/image-projects/vehicle-detection-raspberry-pi-ai-kit#id-3.-model-optimization) for using npy dataset.

**Comment by wb666greene** [comment]
Any reason you can't just use the Ultralytics pre-trained COCO coefficients instead of  redoing it?  Also the "nano" model is fast but it doesn't give very good results.  The "small" is accurate enough to be useful, but the "medium" is the best trade-off for speed-accuracy in my experience. I started back in the days of Ultralytics YOLOv5.

**Reply by LJ-Hao** [reply]
You are right, however, if you want to apply the model to a specific scenario, retraining the pre-trained model is necessary. I would like to demonstrate this process.

**Comment by KunalNN** [comment]
We are having issues doing it with YOLO-v8-pose. any guidience!

**Reply by LJ-Hao** [reply]
Here is the project use yolo pose estimation: [link ](https://github.com/hailo-ai/hailo-rpi5-examples)

**Comment by dje666** [comment]
Are there any missing steps in this tutorial?

$ python3 -m pip install -v -e .

This fails with error: externally-managed-environment......

I create and activate a new virtual environment, but the get warnings about not being in the Hailo virtual environment....?

A few more instructions would be appreciated.

TTFN,

DJE666

**Reply by LJ-Hao** [reply]
Hello, you can use ```source setup_env.sh``` to creat and activate the run time environment, the ```setup_env.sh``` is in this [project](https://github.com/Seeed-Projects/Benchmarking-YOLOv8-on-Raspberry-PI-reComputer-r1000-and-AIkit-Hailo-8L)

**Comment by wxjzyw** [comment]
Hi, when I run hailomz parse --hw-arch hailo8l --ckpt ./best.onnx yolov8n

I am getting the error below. 
miniconda3/envs/hailo_env/lib/python3.10/site-packages/hailo_model_zoo/core/infer/hw_infer_utils.py", line 6, in <module>
    from hailo_platform import (
ModuleNotFoundError: No module named 'hailo_platform'

Is there any step missing ?  Thanks.

**Reply by LJ-Hao** [reply]
If you use yolo model, I suggest you use this [tutorial](https://seeed-projects.github.io/Tutorial-of-AI-Kit-with-Raspberry-Pi-From-Zero-to-Hero/docs/Chapter_5-Custom_Model_Development_and_Deployment/Convert_Your_Model), it is easy for beginner. And if you want use this wiki make sure use ```python -m venv hailo_env``` to creat the environment.
